### PR TITLE
Add and update titles, descriptions in values schema

### PR DESCRIPTION
**Pull request by marians**
Towards https://github.com/giantswarm/roadmap/issues/2096

Here we are adding annotations (title, description, sometimes examples) to the schema to help understand all properties.

### Checklist

- [x] Update changelog in CHANGELOG.md.
- [x] Make sure `values.yaml` and `values.schema.json` are valid.
- [ ] Update Lastpass values if required.

### CODE REVIEW

File: helm/cluster-cloud-director/values.schema.json [hunk 1]
```diff
@@ -87,39 +87,56 @@
...
         },
         "cloudDirector": {
             "properties": {
                 "org": {
+                    "description": "VCD organization name.",
+                    "title": "Organization",
                     "type": "string"
                 },
                 "ovdc": {
+                    "description": "TODO.",
```
Comment (marians) [inline helm/cluster-cloud-director/values.schema.json:128]: The description should clarify what is expected here when filling in the value. Should this be the name or ID of an OvDC the cluster gets created in?

File: helm/cluster-cloud-director/values.schema.json [hunk 2]
```diff
@@ -129,46 +146,62 @@
...
                 "oneArm": {
+                    "description": "TODO.",
                     "properties": {
                         "enabled": {
+                            "title": "Enable",
                             "type": "boolean"
                         }
-                    }
+                    },
+                    "title": "TODO"
```
Comment (marians) [inline helm/cluster-cloud-director/values.schema.json:167]: What is `oneArm`? Thanks for an explanation.

File: helm/cluster-cloud-director/values.schema.json [hunk 3]
```diff
@@ -129,46 +146,62 @@
...
         },
         "cluster": {
             "properties": {
                 "parentUid": {
+                    "description": "If set, create the cluster from a specific management cluster associated with this UID.",
+                    "title": "Management cluster UID",
                     "type": "string"
                 },
                 "rdeId": {
+                    "description": "TODO",
```
Comment (marians) [inline helm/cluster-cloud-director/values.schema.json:185]: What is the `rdeId`?

File: helm/cluster-cloud-director/values.schema.json [hunk 4]
```diff
@@ -523,43 +642,53 @@
...
         },
         "proxy": {
+            "description": "Whether/how outgoing traffic is routed through proxy servers.",
             "properties": {
                 "enabled": {
+                    "title": "Enable",
                     "type": "boolean"
                 },
                 "secretName": {
+                    "title": "TODO",
```
Comment (marians) [inline helm/cluster-cloud-director/values.schema.json:652]: What is `secretName`? What's in that secret?

**Comment by marians** [comment]
@giantswarm/team-rocket I have some questions here regarding the meaning of some properties. See the inline comments.

File: helm/cluster-cloud-director/values.schema.json (hunk 1 above)
Comment (vxav) [reply helm/cluster-cloud-director/values.schema.json:128]: This is the name of the ovdc.

File: helm/cluster-cloud-director/values.schema.json (hunk 2 above)
Comment (vxav) [reply helm/cluster-cloud-director/values.schema.json:167]: oneArm defines whether to use an internal IP for the virtual service with a NAT rule to expose the external IP (true) or to expose the virtual service directly with the external IP (false).
Better explained [here](https://github.com/giantswarm/cloud-provider-cloud-director-app#notes-on-enablevirtualservicesharedip).

File: helm/cluster-cloud-director/values.schema.json [hunk 5]
```diff
@@ -129,46 +146,62 @@
...
-                    }
+                    },
+                    "title": "TODO"
                 }
             },
             "required": [
                 "enableVirtualServiceSharedIP",
                 "oneArm"
             ],
+            "title": "TODO",
```
[the remark this answers is not in the document]
Comment (vxav) [reply helm/cluster-cloud-director/values.schema.json:174]: CPI configuration interface?

File: helm/cluster-cloud-director/values.schema.json (hunk 3 above)
Comment (vxav) [reply helm/cluster-cloud-director/values.schema.json:185]: This is the cluster entity ID in the VCD API.

File: helm/cluster-cloud-director/values.schema.json (hunk 4 above)
Comment (vxav) [reply helm/cluster-cloud-director/values.schema.json:652]: This is the name of the secret where the http_proxy, https_proxy and noproxy environment variables are stored. Used to configure containerd.

**Comment by vxav** [comment]
I think I replied to all of them, let me know if you need more details 🙂 
Thanks for improving this !

**Comment by marians** [comment]
@vxav Thanks for the input! I did my best to integrate it in a readable fashion. Would be happy about a review then.

**Review by vxav — APPROVED** [review]

**Pull request merged**